src/datachain/catalog/catalog.py

```diff
@@ -1198,40 +1252,77 @@ def get_remote_dataset(
 
     def get_dataset_dependencies(
         self,
-        name: str,
-        version: str,
+        name: str | None = None,
+        version: str | None = None,
         namespace_name: str | None = None,
         project_name: str | None = None,
         indirect=False,
+        dataset_id: int | None = None,
+        dataset_version_id: int | None = None,
     ) -> list[DatasetDependency | None]:
-        dataset = self.get_dataset(
-            name,
-            namespace_name=namespace_name,
-            project_name=project_name,
-        )
+        if (name and dataset_id) or (version and dataset_version_id):
+            raise ValueError(
+                "Invalid arguments: only one set of identifiers should be supplied. "
+                "Provide either (name and version),(dataset_id and dataset_version_id)"
+            )
 
-        direct_dependencies = self.metastore.get_direct_dataset_dependencies(
-            dataset, version
+        if not dataset_id:
+            if not name or not version:
+                raise ValueError(
+                    "name and version must be provided when dataset_id is not provided"
+                )
```

Comment:
Just wonder if it will be a good idea to split this function into two (one call the other), for simpler code, less checks, better function signature and typings? 🤔